### ISSUE TITLE
Integration Error on HA Startup

### ISSUE THREAD

**Issue by Krojack** [uvjim/hass_hdhomerun#91]
After May's HA update I've been getting the following error when HDHomerun Integration starts. I have even removed and re-added it.

> 2023-05-05 13:52:41.058 ERROR (MainThread) [homeassistant.config_entries] Error setting up entry HDHomerun Flex 4k for hdhomerun
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/config_entries.py", line 387, in async_setup
>     result = await component.async_setup_entry(hass, self)
>   File "/config/custom_components/hdhomerun/__init__.py", line 134, in async_setup_entry
>     hass.config_entries.async_setup_platforms(config_entry, setup_platforms)
> AttributeError: 'ConfigEntries' object has no attribute 'async_setup_platforms'

I can't think of anything else outside of reinstalling it but I tried that. My HDHomerun 4k Flex still works. Watching stations though Plex everyday.

Current HA Version Info:

> Home Assistant 2023.5.1
> Supervisor 2023.04.1
> Operating System 10.1
> Frontend 20230503.2 - latest

**Comment by uvjim** [comment]
Are you sure you're using the latest version of the integration?  I mitigated this back in August 2022.  The latest version of this integration is 2023.4.1.

**Comment by Krojack** [comment]
I'm using what seems to be built into HA. I go to integration > Add.... Search for it, install, input my devices IP and give it a friendly name. I instantly get that error message. The menifest.json list 2022.3.1.

I manually deleted the custom component, restarted, added this repo to HACS custom, re-installed that way and it's working.

There are now 2 custom integrations that list in HACS. I assumed this was built into HA and would be updated when HA was updated.
